### PR TITLE
Training2

### PR DESCRIPTION
**Pull request by tantrungnguyen2001**
upload training2

### CODE REVIEW

File: Training1/training/src/Questions.js [hunk 1]
```diff
@@ -0,0 +1,9 @@
+export default function Questions(props) {
```
Comment (vanbui1995) [inline Training1/training/src/Questions.js:1]: We should name component as QuestionItem or Question

File: Training1/training/src/Questions.js [hunk 2]
```diff
@@ -0,0 +1,9 @@
...
+  return (
+    <div>
+      <p>Question {props.index}</p>
```
Comment (vanbui1995) [inline Training1/training/src/Questions.js:4]: We should group the appropriate props

eg:
const { question, index } = props;

question.q
question.a

File: Training2/src/task1/src/Component/Modal/Modal.jsx [hunk 3]
```diff
@@ -0,0 +1,31 @@
+import { useState } from "react"
+import './Modal.css'
+
+export default function Modal(){
+    const [modal, setModal] = useState(false)
```
Comment (vanbui1995) [inline Training2/src/task1/src/Component/Modal/Modal.jsx]: the variable name can make misunderstanding, boolean variable should start with is, should...

isOpenModal, setIsOpenModal

File: Training2/src/task1/src/Component/Modal/Modal.jsx [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+    }  
+
+    return (
+        <>
+        <button 
+            id='modal__btn' 
+            onClick={toggleModal}
+        >Press me</button>
+    
+        {modal && (
```
Comment (vanbui1995) [inline Training2/src/task1/src/Component/Modal/Modal.jsx]: in the modal concept, we should not unmount whole modal modal inside depending on the toggle variable

the correct way:

- isMounted: this state control the modal inside mounted or not
- isOpenModal: turn on/off modal but just depend on styling, we don't unmount the component once the component mounted
- isMounted = true when initial value of isOpenModal = true or isOpenModal first change to true

File: Training2/src/task1/src/Component/Modal/Modal.jsx (hunk 3 above)
Comment (vanbui1995) [inline Training2/src/task1/src/Component/Modal/Modal.jsx]: this state can be override by parent component (parent-controlled)

File: Training2/src/task2.1/src/Component/Modal/Modal.jsx [hunk 5]
```diff
@@ -0,0 +1,34 @@
+import React, { Component } from 'react'
+import './Modal.css'
+
+export default class Modal extends Component {
```
Comment (vanbui1995) [inline Training2/src/task2.1/src/Component/Modal/Modal.jsx:4]: same in react component

File: Training2/src/task3/src/App.jsx [hunk 6]
```diff
@@ -0,0 +1,56 @@
...
+      alert('Invalid input. Must be a number')
+    }else if(countdown<0){
+      alert('Number must be greater than 0')
+    }else {
+      timeId.current = setInterval(() => {
+        setCountdown(prev => prev - 1)
+      }, 1000)
+      setToggle(!toggle)
+    }
+    return () => clearInterval(timeId.current)
```
Comment (vanbui1995) [inline Training2/src/task3/src/App.jsx]: why do you return a function on the onClick handler, what is your purpose?

File: Training2/src/task3/src/App.jsx [hunk 7]
```diff
@@ -0,0 +1,56 @@
...
+    e.preventDefault()
+    clearInterval(timeId.current)
+    setToggle(!toggle)
+  }
+
+  useEffect(() => {
+    if(countdown < 0){
+      clearInterval(timeId.current)
+      alert('End')
+    }
```
Comment (vanbui1995) [inline Training2/src/task3/src/App.jsx:52]: Do you miss adding setToggle?

File: Training2/src/task3/src/App.jsx [hunk 8]
```diff
@@ -0,0 +1,56 @@
+import { useEffect, useRef, useState } from 'react'
+import './App.css'
+import CountDown from './Component/CountDown/CountDown'
+// import CountdownSet from './Component/CountdownSet/CountdownSet'
+
+function App() {
+  const [countdown, setCountdown] = useState('')
+  const [toggle, setToggle] = useState(true)
```
Comment (vanbui1995) [inline Training2/src/task3/src/App.jsx]: toggle is no meaning, please change it to "isRunningCountdown"

File: Training2/src/task1/src/Component/Modal/Modal.jsx (hunk 3 above)
Comment (tantrungnguyen2001) [reply Training2/src/task1/src/Component/Modal/Modal.jsx]: dạ vâng

**Pull request closed without merging**